pkg/storage/replica_test.go

```diff
@@ -6265,17 +6265,26 @@ func TestBatchErrorWithIndex(t *testing.T) {
 func TestProposalOverhead(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
+	key := roachpb.Key("k")
 	var overhead uint32
 
 	cfg := TestStoreConfig(nil)
 	cfg.TestingKnobs.TestingProposalFilter =
 		func(args storagebase.ProposalFilterArgs) *roachpb.Error {
-			for _, union := range args.Req.Requests {
-				if union.GetInner().Method() == roachpb.Put {
-					atomic.StoreUint32(&overhead, uint32(args.Cmd.Size()-args.Cmd.WriteBatch.Size()))
-					break
-				}
+			if len(args.Req.Requests) != 1 {
+				return nil
+			}
+			req, ok := args.Req.GetArg(roachpb.Put)
+			if !ok {
+				return nil
 			}
+			put := req.(*roachpb.PutRequest)
+			if !bytes.Equal(put.Key, key) {
+				return nil
+			}
+			atomic.StoreUint32(&overhead, uint32(args.Cmd.Size()-args.Cmd.WriteBatch.Size()))
+			args.Cmd.WriteBatch = nil // shallow modification
```

Comment:
What is this about? This addition seems more significant than the commit message indicates.

Reply:
I don't want to `pretty.Sprint` the `WriteBatch` because it's intentionally excluded from the size computation. `args` is a value, and `Cmd` isn't a pointer neither, so we're not changing the data the caller sees, which I was trying to indicate in the comment. Does that make sense?

Reply:
Ah, got it. The comment wasn't helping me there. Perhaps `// avoid printing the WriteBatch (shallow modification)`.

Reply:
I went full verbosity.